### ISSUE TITLE
hashed_variants confusion

### ISSUE THREAD

**Issue by jamesmclaren** [nerdvegas/rez#754]
Hi there,
I appreciate that hashed variants solves a problem with nested folders for variants or variants with difficult characters. However, as far as I can see there is no way by eye to work out which variant is in which folder. What if when the package is created, a text file was created in the folder indicating which variant it was?
Such as a file called variant.txt which could contain the variant string ['platform-windows', 'arch-AMD64', 'python-2.7'].
This would make things easier to manage. 
Could hashed variants be optional? As we have been using the path mapping to simplify our variant paths anyway. I've noted that variant_shortlinks might be a way to create user friendly links, but since are servers use an Isilon cluster, with a mixture of windows and Linux machines, symlinks don't work. 
What do you think?
James

**Comment by instinct-vfx** [comment]
I agree that adding information about the variant in a file would be helpful. Hashed variants are optional. They are controlled with a package attribute though and the default is true. I don't think there is a config attribute to change the default yet.

**Comment by nerdvegas** [comment]
There's already a file for this very reason, it's called "variant.json" and
should be present in the root of each variant.

Also, hashed variants _are_ optional, you need to set "hashed_variants =
True" in your package definition. However they are mandatory in rez-pip (is
that what you meant), because in that case you don't have direct control
over the variants that get created (those depend on a few factors,
including any environment markers that the python package may use).

Hth
A

**Comment by jamesmclaren** [comment]
we are using rez-pip, and as far as I can see it's not creating a variant.json file, that seems to done in LocalBuildProcess._build_variant_base(), which as far as I can see is not called by rez-pip

**Comment by nerdvegas** [comment]
Ah right. Yes in that case you won't get the variant.json, nor do you get
shortlinks. Rez-pip follows a different code path that wasn't really
anticipated when this stuff was done, hence shortlinks/variant.json only
happening via a rez-build. We do need to figure out how to rectify that so
that it happens in both cases.

Wrt hashed variants. Whilst it would be possible to disable them in
rez-pip, I am hesitant. The reason is that, for various reasons, I do see
hashed variants as being how pretty much everything will work in future.
When we have support for public rez package repositories, the way variants
will be constructed will be a lot more automated, and the idea of having
manual control over variant construction won't make much sense.

I would say the thing we should work on first is getting variant.json
written for rez-pip as well, hopefully using code common to rez-build. How
does that sound?

Thx
A

**Comment by jamesmclaren** [comment]
Hi Allan,
Writing the variant.json for rez-pip would be great, as at the moment these folders are difficult to work out what in each one, especially since we are storing Windows and Linux variants in the same folder structure. If you then have to customize a package's contents after rez-piping, it could become awkward.
Thanks
James
